client/asset/dcr/wallet.go

```diff
@@ -92,54 +85,65 @@ type Wallet interface {
-	// mempool.
-	GetRawMempool(ctx context.Context, txType chainjson.GetRawMempoolTxTypeCmd) ([]*chainhash.Hash, error)
+	GetRawTransaction(ctx context.Context, txHash *chainhash.Hash) (*wire.MsgTx, error)
 	// GetBestBlock returns the hash and height of the wallet's best block.
 	GetBestBlock(ctx context.Context) (*chainhash.Hash, int64, error)
 	// GetBlockHash returns the hash of the mainchain block at the specified height.
 	GetBlockHash(ctx context.Context, blockHeight int64) (*chainhash.Hash, error)
-	// BlockCFilter fetches the block filter info for the specified block.
-	BlockCFilter(ctx context.Context, blockHash *chainhash.Hash) (filter, key string, err error)
-	// LockWallet locks the wallet.
-	LockWallet(ctx context.Context) error
-	// UnlockWallet unlocks the wallet.
-	UnlockWallet(ctx context.Context, passphrase string, timeoutSecs int64) error
-	// WalletUnlocked returns true if the wallet is unlocked.
-	WalletUnlocked(ctx context.Context) bool
-	// AccountUnlocked returns true if the specified account is unlocked.
-	AccountUnlocked(ctx context.Context, account string) (*walletjson.AccountUnlockedResult, error)
-	// LockAccount locks the specified account.
-	LockAccount(ctx context.Context, account string) error
-	// UnlockAccount unlocks the specified account.
-	UnlockAccount(ctx context.Context, account, passphrase string) error
+	// BlockFilter fetches the block filter info for the specified block.
+	BlockFilter(ctx context.Context, blockHash *chainhash.Hash) ([gcs.KeySize]byte, *gcs.FilterV2, error)
+	// Unlocked returns true if the Wallet unlocked.
+	Unlocked(ctx context.Context) (bool, error)
+	// Lock locks the Wallet. ExchangeWallet does not differentiate account
+	// locking vs wallet locking, but the underlying implementation may choose
+	// to lock only the account.
+	Lock(ctx context.Context) error
+	// Unlock unlocks the Wallet. ExchangeWallet does not differentiate account
+	// locking vs wallet locking, but the underlying implementation may choose
+	// to unlock only the account.
+	Unlock(ctx context.Context, passphrase []byte) error
 	// SyncStatus returns the wallet's sync status.
 	SyncStatus(ctx context.Context) (bool, float32, error)
 	// PeerCount returns the number of network peers to which the wallet or its
 	// backing node are connected.
 	PeerCount(ctx context.Context) (uint32, error)
 	// AddressPrivKey fetches the privkey for the specified address.
-	AddressPrivKey(ctx context.Context, address stdaddr.Address) (*dcrutil.WIF, error)
+	AddressPrivKey(ctx context.Context, address stdaddr.Address) (*secp256k1.PrivateKey, error)
+}
+
+type FeeRateEstimator interface {
+	// EstimateSmartFeeRate returns a smart feerate estimate.
+	EstimateSmartFeeRate(ctx context.Context, confTarget int64, mode chainjson.EstimateSmartFeeMode) (float64, error)
+}
+
+type Mempooler interface {
+	// GetRawMempool returns hashes for all txs in a node's mempool.
+	GetRawMempool(ctx context.Context) ([]*chainhash.Hash, error)
 }
```

Comment:
I guess the Interfaces themselves could use comments since exported. Do these need to be exported? Will core will use them in the future, or the wallets be split into different packages? The `dcr.Wallet` interface is also exported although I don't think it needs to be.